spec/integration/operations_spec.rb

```diff
@@ -169,34 +176,51 @@ module Hawkular::Operations::RSpec
                            example)
         end
 
-        it 'Add JDBC driver should add the driver' do # Unless it runs in a container
-          driver_name = 'CreatedByRubyDriver' + @not_so_random_uuid
+        def add_jdbc_driver
           driver_bits = IO.binread("#{File.dirname(__FILE__)}/../resources/driver.jar")
           wf_path = CanonicalPath.new(tenant_id: @tenant_id,
                                       feed_id: @feed_id,
-                                      resource_ids: [@wf_server_resource_id]).to_s
+                                      resource_ids: [wf_server_resource_id]).to_s
 
           actual_data = {}
 
           client.add_jdbc_driver(resource_path: wf_path,
                                  driver_jar_name: 'driver.jar',
                                  driver_name: driver_name,
-                                 module_name: 'foo.bar' + @not_so_random_uuid, # jboss module
+                                 module_name: 'foo.bar' + not_so_random_uuid, # jboss module
                                  driver_class: 'com.mysql.jdbc.Driver',
                                  binary_content: driver_bits) do |on|
             on.success do |data|
               actual_data[:data] = data
             end
             on.failure do |error|
               actual_data[:data] = error
-              puts 'error callback was called, reason: ' + error.to_s unless @agent_immutable
             end
           end
-          actual_data = wait_for actual_data
-          expect(actual_data['status']).to eq('OK') unless @agent_immutable
-          expect(actual_data['message']).to start_with('Added JDBC Driver') unless @agent_immutable
-          expect(actual_data['driverName']).to eq(driver_name) unless @agent_immutable
-          expect(actual_data).to include('Command not allowed because the agent is immutable') if @agent_immutable
+          wait_for actual_data
+        end
+
+        context 'Mutable agent' do
+          before(:each) do
+            skip 'mutable agent specs don\'t run on immutable agent' if @agent_immutable
```

Comment:
I still don't like this. We should duplicate the setup to create one mutable and one immutable, if that's what it takes. Tests should be predictable, and any behaviour that introduces variants should be extracted into different tests/different contexts.